pyscitt/pyscitt/verify.py

```diff
@@ -89,12 +99,33 @@ def verify_receipt(
     decoded_receipt.verify(msg, service_params)
 
 
+def verify_transparent_statement(
```

Comment:
Measurement check was removed, test env should generate a mock measurement which must be present in the receipt headers

Reply:
The code measurement of the TEE that accepted the registration is not included in the SCITT drafts. It's something that can be obtained from the CCF ledger by an auditor if desired.

Reply:
This is part of the feature we added which allows us to show how to reproduce the specific version that issued the receipt. It is not about SCITT drafts because they do not really address reproducibility either. There is no version information in the TLS cert as well, so how does the user know which version of code issued the receipt? The suggestion to look at the ledger files is suboptimal because there is no viable path to it at the moment, AFAIK it exists in theory only.

Reply:
I do not know how to resolve the exact version which issued the receipt, if I miss something then please expand because this is an open issue to make sure SEV-SNP is reproducible and traceable. We need to have concrete steps to link the receipt to the measurement, please update `docs/reproducibility.md` with the steps necessary to link those.

Reply:
> The suggestion to look at the ledger files is suboptimal because there is no viable path to it at the moment, AFAIK it exists in theory only.

Do feel free to propose an alternative. I will continue to focus updating the implementation to match the IETF specs.

> We need to have concrete steps to link the receipt to the measurement, please update docs/reproducibility.md with the steps necessary to link those.

This is out of scope for this PR and for this branch.

Reply:
> This is out of scope for this PR and for this branch.

The measurement is being removed in this PR so it should be in scope, I do not understand what is your suggestion, just to forget about it? Is it not possible to update the doc?

Reply:
The scope of this PR (which is to a dev branch, not to main) is to switch CTS to COSE receipts. Many things are possible, but are out of scope, including updating the documentation.

To bring some clarity on the topic of having nodes sign their TEE measurement: this never created a security property. The only valuable signature over these measurements is that contained in the hardware attestation.

A relying party can establish all they need to know about the TEE from the attestation, and from the attestation alone, an additional signature by a key generated by the code running inside the TEE creates no additional property. Signing the TEE measurement with a key created by the code is wasteful at best, and dangerous at worst because relying parties may interpret it as an authoritative statement about the code that signed the ledger, which it is not.

Even the node attestation is insufficient to replay/audit a single signature transaction, because the code is only one of the inputs, the state of the ledger/KV is the other. The state of the ledger (captured in the Merkle Proof) contains all past node attestations, and is always sufficient for audit purposes. Extracting an unauthenticated measurement from it does not create a additional security property.

Finally, a node attestation has limited usefulness for audit, even with node-scoped keys, without establishing the history of the service key used to endorse it. This is particularly true for a pure-integrity application like CTS, which aims to produce a single, unequivocal history. Adding an unauthenticated measurement to signatures neither improves the full audit story, nor partial consistency checks based on a set of receipts.